### DISCUSSION TITLE
How to update `nix-on-droid`?

### DISCUSSION THREAD

**Discussion by auroraanna** [nix-community/nix-on-droid#172] [Q&A]
I want to switch to a flake config but `nix-on-droid switch --flake ...` says that the option `--flake` is unknown. It is in the readme though so my binary must be out of date. How to update it?

**Comment by t184256** [comment]
What's in your `nix-channel --list`?

**Comment by auroraanna** [comment]
```
bash-5.1$ nix-channel --list
nix-on-droid https://github.com/t184256/nix-on-droid/archive/release-21.05.tar.gz
nixpkgs https://nixos.org/channels/nixpkgs-unstable
```

**Comment by t184256** [comment]
Uh-huh. `nix-channel --add https://github.com/t184256/nix-on-droid/archive/release-21.11.tar.gz nix-on-droid` and `nix-channel --update`, then you should have it.

**Reply by auroraanna** [reply]
That didn't fix it. The option is still not there.

**Reply by Gerschtli** [reply]
You need to run nix-on-droid switch once after updating the channel to be able to use the option. Currently, you are still using the old version of nix-on-droid. It behaves the same like nixos-rebuild, nix-env, home-manager etc.

**Reply by t184256** [reply]
Yeah, right, `nix-on-droid switch` too.
